### ISSUE TITLE
Constructor as.ejObject not working - IDs issue

### ISSUE THREAD

**Issue by thibautjombart** [Hackout2/repijson#18]
Not sure if this relates to https://github.com/Hackout2/repijson/issues/16, but, from `?toyll`:

``` r
 data(toyll)
 toyll

ind.fields <- c(names(toyll)[1:5], "hospital", "fever", "sleepy")

x <- as.ejObject(toyll,
                   recordAttributes=ind.fields,
                   eventDefinitions=list(
                   define_ejEvent(name="admission", date="date.of.admission"),
                   define_ejEvent(name="discharge", date="date.of.discharge"),
                   define_ejEvent(name="contact1", date="contact1.date", attributes="contact1.id"),
                   define_ejEvent(name="contact2", date="contact2.date", attributes="contact2.id"),
                   define_ejEvent(name="contact3", date="contact3.date", attributes="contact3.id")
                   ))

```

I get:

``` r
Error in create_ejEvent(id = notNA(rd$id, x[i, rd$id]), name = notNA(rd$name,  : 
  event id must be numeric
```

@EllieSherrardSmith and I will work on the examples as soon as this is fixed.
@twomagpi @AndySouth any idea?

**Comment by twomagpi** [comment]
Hi,
I have a couple of commits that may address this. I’ll push them in a bit (they just break other things so I was going to push all in one go)
Tom

**Comment by andysouth** [comment]
Good to see progress on this. I'm on holiday next week so will check back in the week after.

**Comment by twomagpi** [comment]
Fixed as of 33bf45d948d21e5108cfbf06479f810c8379b473
